draft-romann-mud-constrained.md

```diff
@@ -214,20 +214,72 @@ Using additional Link-Format parameters and well-known URIs, this document
 introduces new possibilities for discovering MUD URLs in constrained
 environments.
 
-## MUD CoAP Payloads
-
-- Text/Plain
-    - text/mud-url+plain
-- CBOR (unsigned)
-    - application/mud-url+cbor
-- Signed MUD-URL Objects
-    - application/mud-url+cose...?
-
 ## MUD-URL CoAP Submission Flows
+In general, this specification provides two ways by which a MUD-URL transmission can be performed using CoAP.
+In environments where many Things need to be managed over several subnets and where multicast usage is not desirable, it can be advantageous if the MUD receiver provides a CoAP resource to perform submissions to and the Things initiate the MUD-URL submission.
+Conversely, in environments where multicast is not an issue and things might be limited in their capabilities, it can be advantageous if MUD receivers retrieve the MUD-URL from a CoAP resource provided by the Things.
+
+### Using the MUD-URL Resource (Receiver-initiated)
+In the Receiver-initiated flow, Things provide a CoAP resource discoverable by the means provided in [REF] <!-- TODO REF resource discovery -->, which is then requested by MUD receivers to retrieve the MUD-URL.
+
+<!-- TODO ascii-drawing of this resource -->
+
+In general, the Receiver-initiated MUD-URL flow can be divided into these steps:
+1.  After joining the network, the Thing starts providing a CoAP resource to retrieve the MUD-URL.
+    This resource should provide the MUD-URL in one of the formats specified in [REF] <!-- TODO REF MUD CoAP Payloads -->.
+    It also makes this resource discoverable for MUD receivers using the methods specified in [REF] <!-- TODO REF resource discovery -->. 
+2.  The MUD Receiver discovers the resource using the aforementioned methods.
+    Depending on the method of discovery, this could for example happen using a periodic scan for devices, e.g., by periodically requesting a well-known URI using multicast.
+    Other methods of discovery might also provide a mechanism to directly notify the Receiver of new devices, in which case this method SHOULD be preferred over periodic scanning.
+3.  The MUD Receiver retrieves the discovered resource for devices where the MUD controller does not have an up-to-date MUD-URL stored.
+    To do so, it performs a CoAP request for the discovered MUD-URL resource URI using the GET method, which is responded to with the appropriate payload.
+    Receivers MUST specify the desired payload formats using the Accept option <!-- TODO cite -->.
+    If multiple payload formats are specified, the Thing MUST prefer ones that provide a greater degree of authenticity protection (i.e., prefer CWTs over plaintext transmission).
```

Comment:
The Accept option is actually non-repeatable, so from my understanding, content negotiation would occur over multiple exchanges if the Thing cannot fulfill the initial request. Would something like this work instead?

```suggestion
    Receivers MUST specify their desired payload format using the Accept option {{Section 5.10.4 of RFC7252}}.
    During content negotiation, Receivers MUST start with requesting the Content-Format that provides the greatest degree of authenticity protection (i.e., prefer CWTs over plaintext transmission).
```